File: app/src/androidTest/java/com/github/jameshnsears/quoteunquote/utils/ToastHelperTest.java
package com.github.jameshnsears.quoteunquote.utils;

import android.widget.Toast;

import com.github.jameshnsears.quoteunquote.report.ActivityReport;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import androidx.test.annotation.UiThreadTest;
import androidx.test.ext.junit.runners.AndroidJUnit4;
import androidx.test.rule.ActivityTestRule;

import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertNull;

@RunWith(AndroidJUnit4.class)
public class ToastHelperTest {
    @Rule
    public ActivityTestRule<ActivityReport> activityRule =
            new ActivityTestRule(ActivityReport.class);

    @Test
    @UiThreadTest
    public void makeToast() {
        assertNull("", ToastHelper.toast);

        ToastHelper.makeToast(
                activityRule.getActivity().getApplicationContext(),
                "msg",
                Toast.LENGTH_SHORT);

        assertNotNull("", ToastHelper.toast);
    }
}

File: utilsLib/src/test/java/com/github/jameshnsears/quoteunquote/utils/ui/ToastHelperTest.kt
package com.github.jameshnsears.quoteunquote.utils.ui

import android.content.Context
import android.os.Build
import android.widget.Toast
import androidx.test.core.app.ApplicationProvider.getApplicationContext
import androidx.test.ext.junit.runners.AndroidJUnit4
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.annotation.Config

@RunWith(AndroidJUnit4::class)
@Config(sdk = [Build.VERSION_CODES.P])
class ToastHelperTest {
    @Test
    fun `demonstrate invocation`() {
        val context: Context = getApplicationContext()
        ToastHelper.makeToast(context, "message1", Toast.LENGTH_LONG)
        ToastHelper.makeToast(context, "message2", Toast.LENGTH_SHORT)
    }
}
